File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Parameters/Parameter.cs
﻿using System;
using VecompSoftware.DocSuiteWeb.Entity.Tenants;

namespace VecompSoftware.DocSuiteWeb.Entity.Parameters
{
    public class Parameter : DSWBaseEntity
    {
        #region [ Constructor ]
        public Parameter() : this(Guid.NewGuid()) { }
        public Parameter(Guid uniqueId)
            : base(uniqueId) { }
        #endregion

        #region [ Properties ]

        public int Incremental { get; set; }
        public short LastUsedYear { get; set; }
        public int LastUsedNumber { get; set; }
        public bool Locked { get; set; }
        public short LastUsedIdCategory { get; set; }
        public short LastUsedIdContainer { get; set; }
        public short LastUsedIdRole { get; set; }
        public short LastUsedIdRoleUser { get; set; }
        public int? LastUsedIdResolution { get; set; }
        public short LastUsedResolutionYear { get; set; }
        public short LastUsedResolutionNumber { get; set; }
        public short LastUsedBillNumber { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual TenantAOO TenantAOO { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Common/Parameter.vb
<Serializable()> _
Public Class Parameter
    Inherits AuditableDomainObject(Of Integer)
    Implements ISupportTenant


#Region "private data"

#End Region

#Region "Properties"

    Public Overridable Property LastUsedResolutionYear As Short
    Public Overridable Property LastUsedResolutionNumber As Short
    Public Overridable Property LastUsedBillNumber As Short
    Public Overridable Property LastUsedYear As Short
    Public Overridable Property LastUsedNumber As Integer
    Public Overridable Property Locked As Boolean
    Public Overridable Property LastUsedIdCategory As Short
    Public Overridable Property LastUsedIdContainer As Short
    Public Overridable Property LastUsedIdResolution As Integer
    Public Overridable Property LastUsedIdRole As Short
    Public Overridable Property LastUsedIdRoleUser As Short
    Public Overridable Property IdTenantAOO As Guid? Implements ISupportTenant.IdTenantAOO

#End Region

#Region "Ctor/init"
    Public Sub New()

    End Sub
#End Region

    End Class

